core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -136,7 +135,7 @@
     private DataSize filterAndProjectMinOutputPageSize = DataSize.of(500, KILOBYTE);
     private int filterAndProjectMinOutputPageRowCount = 256;
     private int maxGroupingSets = 2048;
-    private JoinPushdownMode joinPushdownMode = JoinPushdownMode.OFF;
+    private JoinPushdownMode joinPushdownMode = JoinPushdownMode.EAGER;
```

Comment:
Makes sense. Changing.

Reply:
please still add a code comment why not EAGER

Reply:
@findepi still want that given https://github.com/trinodb/trino/pull/6818 changes that to `AUTOMATIC`?

Reply:
Yes, I think we should record the reason why EAGER is not made the default, or why we did not implement AUTOMATIC as "AUTOMATIC or EAGER", and have that as the default.

Ideally the addition of the comment should go before the commit switching the default, so that it remains clear that we're commenting the state as currently merged in master, not a state after #6818 .